src/main/java/io/numaproj/numaflow/function/MessageT.java

```diff
@@ -0,0 +1,36 @@
+package io.numaproj.numaflow.function;
+
+import static io.numaproj.numaflow.function.Message.ALL;
+import static io.numaproj.numaflow.function.Message.DROP;
+
+import java.time.Instant;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/**
+ * MessageT is used to wrap the data return by UDF functions. Compared with Message, MessageT
+ * contains one more field, the event time, usually extracted from the payload.
+ */
+@AllArgsConstructor
+@Getter
+public class MessageT {
+
+    private Instant eventTime;
```

Comment:
This could be any time right? a user could choose to have it as current time or adjust the time to account for lateness etc? Is that a valid usecase for the source transformer?

Reply:
Yes this could be any time. User can implement the transformer to adjust the time. The main purpose is to provide a way to extract event time from payload but technically user can choose to abuse this functionality. See https://github.com/numaproj/numaflow/issues/446 issue description: Benefits/Watermark Assignment